dbcon/joblist/distributedenginecomm.cpp

```diff
@@ -1054,7 +1054,10 @@ int DistributedEngineComm::writeToClient(size_t index, const ByteStream& bs, uin
         alarmItem.append(" PrimProc");
         alarmMgr.sendAlarmReport(alarmItem.c_str(), oam::CONN_FAILURE, SET);
         */
-        throw runtime_error("DistributedEngineComm::write: Broken Pipe error");
+
+        // Re-establish connection before throwing error 
+        Setup();
```

Comment:
I wonder if Setup() can throw on its own? Control flow will go into another exceptions handler so we'll lost the error message and luckily if the new exception also belongs to runtime_error class.